packages/async-flow/test/async-flow-wake.test.js

```diff
@@ -0,0 +1,217 @@
+// The purpose of this test file is to demonstrate various wake behavior: if a
+// flow is either an eager waker or in a failed state it should be awoken in a
+// new incarnation, but otherwise should wait until a wake watcher is triggered.
+
+import { Fail } from '@endo/errors';
+import { testAsyncLife } from './prepare-test-env-ava.js';
```

Comment:
I suggest there could be an initial `testAsyncLife`, which returns a new `testAsyncLife` with its reincarnation state curried in:

```suggestion
import { testAsyncLifeFromScratch } from './prepare-test-env-ava.js';
```